### PR TITLE
User-Agent FormatException: Invalid HTTP header field value

### PR DESCRIPTION
**Pull request by litang0908**
API request failed if  there are Chinese characters in User Agent. so encode ua by `Uri.encodeComponent`.

Here is the error message:
```
FormatException: Invalid HTTP header field value: "MacBookPro18,3, Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000, 版本13.5.1（版号22G90）" (at character 120)
...2:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000, 版本13.5.1（版号22G90）
```

### CODE REVIEW

File: lib/src/matomo.dart [hunk 1]
```diff
@@ -352,6 +352,15 @@ class MatomoTracker {
   @visibleForTesting
   Future<String?> getUserAgent({
     DeviceInfoPlugin? deviceInfoPlugin,
+  }) async {
+    String? result =
+        await _getUserAgentInner(deviceInfoPlugin: deviceInfoPlugin);
+    return result != null ? Uri.encodeComponent(result) : null;
+  }
+
+  @visibleForTesting
```
Comment (TesteurManiak) [inline lib/src/matomo.dart:361]: Do not keep the annotation `visibleForTesting` on a private method.

File: lib/src/matomo.dart [hunk 2]
```diff
@@ -352,6 +352,15 @@ class MatomoTracker {
...
+  Future<String?> _getUserAgentInner({
+    DeviceInfoPlugin? deviceInfoPlugin,
```
Comment (TesteurManiak) [inline lib/src/matomo.dart:363]: You don't have to use a named parameter in a private method.

File: lib/src/matomo.dart [hunk 3]
```diff
@@ -352,6 +352,15 @@ class MatomoTracker {
   @visibleForTesting
   Future<String?> getUserAgent({
     DeviceInfoPlugin? deviceInfoPlugin,
+  }) async {
+    String? result =
+        await _getUserAgentInner(deviceInfoPlugin: deviceInfoPlugin);
+    return result != null ? Uri.encodeComponent(result) : null;
```
Comment (TesteurManiak) [inline lib/src/matomo.dart:358]: Prefer the syntax:
```dart
final result = await _getUserAgentInner(deviceInfoPlugin);
return result != null ? Uri.encodeComponent(result) : null;
```

**Review by TesteurManiak — CHANGES_REQUESTED** [review]
Thanks for the contribution, those feedbacks are mostly related to the coding style to keep a consistent codebase.

**Review by Pierre-Monier — COMMENTED** [review]
I agree with @TesteurManiak, those changes will also make the CI happy

File: lib/src/matomo.dart (hunk 3 above)
Comment (litang0908) [reply lib/src/matomo.dart]: Thanks for these guides, i have reorganized the code.

**Comment by TesteurManiak** [comment]
@litang0908 Some tests are failing and should be fixed before merging.

**Pull request closed without merging**